### PR TITLE
Adding ctest for the new amsr2_gcom-w1 converter

### LINKED ISSUE #1332: amsr2 converter updates
## Current behavior (describe the bug)

testing end-to-end through skylab found some small cleanup to be done

the channel numbers started with 0 rather than 1
had to double check the order of the horizontal and vertical polarization channels
various scaling factors be applied correctly for factors
ensure satellite and solar angles are computed correctly

### PR DESCRIPTION
**Pull request by smaticka**
## Description

A ctest of the down-sampled `amsr2_gcom-w1` data needs to be added to confirm the `ioda-converter` works as it was designed.

## Issue(s) addressed

Resolves #1332 

## Dependencies

PR #1333 - to fix the `amsr2` converter 

## Checklist

- [x] add down-sampled raw data in `test/testinput`
- [x] add down-sampled converted data in `test/testoutput`
- [x] add the `ctest` information for `amsr2` in `ioadconv/test/CMakeList.txt`
- [x] add converter to list which copies over `amsr2_gcom-w1.py` into the `build/bin/` directory.

### CODE REVIEW

**Comment by PatNichols** [comment]
@smaticka Does this PR depend on #1333 ? Please put that in the description of the PR if so. Thank you!

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -891,6 +893,18 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_sondetac
...
+                  TYPE    SCRIPT
+                  ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
+                  COMMAND bash
+                  ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                          netcdf
+                  "${CMAKE_BINARY_DIR}/bin/amsr2_2ioda.py
+                  -i testinput/amsr2_gcom-w1_20220216.h5
+                  -o testrun/amsr2_gcom-w1_20220216.nc4
+                  -d 2022021600"
+                  amsr2_gcom-w1_20220216.nc4 ${IODA_CONV_COMP_TOL_ZERO} )
```
Comment (PatNichols) [inline test/CMakeLists.txt:906]: @smaticka judging from the ctests this probably too tight of a tolerance.

Comment (smaticka) [reply test/CMakeLists.txt:906]: What does tolerance mean? the size of the files?

Comment (smaticka) [reply test/CMakeLists.txt:906]: all tests passed for me

**Comment by smaticka** [comment]
@PatNichols It looks like this PR and the GMI ctest PR (#1339 ) are both not passing the the same check  `AWS CodeBuild us-east-1 (ioda-conventers-internal-intel)`. Do you know what the problem might be? they are very similar code edits. I found an extra space in the CMakeList.txt file in each PR and removed them, but they are both still failing.

File: test/CMakeLists.txt (hunk 1 above)
Comment (PatNichols) [reply test/CMakeLists.txt]: It only happened on intel compilers. Looks like a floating point thing. @fcvdb will talk to you about what's needed.

**Comment by PatNichols** [comment]
@smaticka the tolerance variable is set to zero. The definition is in the cmakelists.txt. you can change it from zero to say 1e-4 and see if passes

**Comment by smaticka** [comment]
> @smaticka the tolerance variable is set to zero. The definition is in the cmakelists.txt. you can change it from zero to say 1e-4 and see if passes

Great, thanks! I just made the change to try it

**Comment by smaticka** [comment]
Thanks for the tip @PatNichols ! Looks like all checks are passing now and the PR is ready for review

**Review by BenjaminRuston — CHANGES_REQUESTED** [review]

**Review by BenjaminRuston — APPROVED** [review]
perfect thanks @smaticka

**Review by fcvdb — APPROVED** [review]
Thanks Sam!

**Review by PatNichols — APPROVED** [review]
@smaticka Thank you for doing this.

**Pull request merged**